pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1316,31 +1316,18 @@ func (og *operationGenerator) GenerateUnmapDeviceFunc(
 		}
 
 		// The block volume is not referenced from Pods. Release file descriptor lock.
-		// This should be done before calling TearDownDevice, because some plugins that do local detach
-		// in TearDownDevice will fail in detaching device due to the refcnt on the loopback device.
-		klog.V(4).Infof("UnmapDevice: deviceToDetach.DevicePath: %v", deviceToDetach.DevicePath)
-		loopPath, err := og.blkUtil.GetLoopDevice(deviceToDetach.DevicePath)
+		// This should be done before calling UnmapDevice, because some plugins that do local detach
+		// in UnmapDownDevice will fail in detaching device due to the refcnt on the loopback device.
+		err = og.blkUtil.DetachFileDevice(deviceToDetach.DevicePath)
```

Comment:
so for CSI, this call will do nothing, is that right. Since CSI did AttachFileDevice for each of its publish paths, rather than at the deviceToDetach.DevicePath...

Reply:
NVM, we discussed more in detail above the purpose of DetachFileDevice.